### PR TITLE
va-search-input: Add disableAnalytics check for submit and blur callbacks

### PR DESCRIPTION
**Pull request by randimays**
## Chromatic

https://VACMS-18846-search-input-analytics--65a6e2ed2314f7b8f98609d8.chromatic.com

---
## Configuring this pull request
- [x] Link to any related issues in the description so they can be cross-referenced.
- [x] Add the appropriate version patch label (`major`, `minor`, `patch`, or `ignore-for-release`).
    - See [How to choose a version number](https://github.com/department-of-veterans-affairs/component-library#how-to-choose-a-version-number) for guidance.
    - Use `ignore-for-release` if files haven't been changed in a component library package. (ie. only Storybook files)
- [ ] DST Only: Increment the `/packages/core` version number if this will be the last PR merged before a release.
- [ ] Complete all sections below.
- [ ] Delete this section once complete

## Description
Closes [3122](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3122).

Public Websites' companion ticket: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18846

## QA Checklist
- [x] Component maintains 1:1 parity with design mocks
- [ ] Text is consistent with what's been provided in the mocks
- [ ] Component behaves as expected across breakpoints
- [ ] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [ ] Tab order and focus state work as expected
- [ ] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [x] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [ ] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Acceptance criteria
- [ ] QA checklist has been completed
- [ ] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by jamigibbs** [comment]
**Tested with Verdaccio in vets-website**

- onblur and onsubmit

`disable-analytics`

https://github.com/user-attachments/assets/f1078850-a6c6-4714-8602-26e6b45759dd

`disable-analytics = false`

https://github.com/user-attachments/assets/c54aa1e3-0c2f-4a76-a6f6-59d918ec5535

**Comment by randimays** [comment]
Thanks for your help with testing, @jamigibbs!

File: packages/web-components/src/components/va-search-input/test/va-search-input.e2e.ts [hunk 1]
```diff
@@ -841,6 +841,36 @@ describe('va-search-input', () => {
     });
   });
 
+  it('does not fire an analytics event when button is clicked if analytics are disabled', async () => {
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-search-input/test/va-search-input.e2e.ts:844]: These tests looks good. Can you also add tests to check if setting the prop to `true` works as expected as well?

**Review by jamigibbs — COMMENTED** [review]
Thanks for working on this! I just have a quick request to beef up the tests a little bit.

File: packages/web-components/src/components/va-search-input/test/va-search-input.e2e.ts (hunk 1 above)
Comment (randimays) [reply packages/web-components/src/components/va-search-input/test/va-search-input.e2e.ts]: Sure, just added!

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**